saos-search/src/main/java/pl/edu/icm/saos/search/indexing/ReindexJobStepExecutionListener.java

```diff
@@ -35,6 +44,14 @@
     public void beforeStep(StepExecution stepExecution) {
         SourceCode sc = EnumUtils.getEnum(SourceCode.class, sourceCode);
         judgmentRepository.markAsNotIndexedBySourceCode(sc);
+        
+        if (sc == null) {
```

Comment:
Yes, it would be more consistent.
But on the other side we added deleting before reindex to prevent errors when something has changed in index. When we create query to delete judgments with specified sourceCode we are depending on fact that there is field 'sourceCode' and nothing has changed with it after schema.xml updates.
Is this something that we can accept?

One more thing. We save in index only courtType. Is it okay to map SourceCode to CourtType or we must put SourceCode into index as well?